### PR TITLE
Add projectDependencyGraph Gradle

### PR DESCRIPTION
**Pull request by michalharakal**
## Description
Add a new Gradle task to the project, which generates image with dependency graph from the project modules structure. Implementation by Jake Wharton https://github.com/JakeWharton/SdkSearch/blob/master/gradle/projectDependencyGraph.gradle

## Type of change

Please delete options that are not relevant.

-   [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested

-   [ ] Readme file contains a new generated picture

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: scripts/gradle/projectDependencyGraph.gradle [hunk 1]
```diff
@@ -0,0 +1,116 @@
+task projectDependencyGraph {
```
Comment (VMadalin) [inline scripts/gradle/projectDependencyGraph.gradle]: All the `gradle` files are migrated to `kotlin dsl` in order to keep project consistency maybe we should do the same with this one. I found an already [one](https://github.com/vanniktech/gradle-dependency-graph-generator-plugin/blob/1c065cb50041b911effb6b2c61406242a33ac8d4/src/main/kotlin/com/vanniktech/dependency/graph/generator/ProjectDependencyGraphGenerator.kt),  as a good start point.

The path should be `buildSrc/.../plugins` or `buildSrc/.../utils` if you consider to migrate to kotlin. The `scripts` directory was created for bash scripts not for gradle tasks.

**Comment by VMadalin** [comment]
First of all, Thanks so much for your changes :).  I appreciate your very helpful contribution and I left a comment regards project style.  

PS: I also improve the project PR checks.

File: scripts/gradle/projectDependencyGraph.gradle (hunk 1 above)
Comment (michalharakal) [reply scripts/gradle/projectDependencyGraph.gradle]: OK, Good point, I was already looking for one, I didn't find that. I will check it

**Comment by VMadalin** [comment]
Apparently @michalharakal your changes break the build due having pushed this ignored file `app/fabric.properties` to the repo with unexpected values. Overriding all the entire build logic related with create this file using CI values for that.

**Comment by michalharakal** [comment]
Sorry for chaos, feel free do anything usefull (close PR) to clean the stuff. I try provide a new PR soon. I am still learning a lot from your work. I still want to implement kotlin based graphiviz generator as plugin.

**Comment by VMadalin** [comment]
Don't worry Michal :) I appreciate your work and of course your idea to create this graph automatically is awesome.  Feel free to ask me any question, doubt regarding the project to try to help you, by creating an issue or if you prefer via slack.

**Pull request closed without merging**